### ISSUE TITLE
Discussion Episode 16: Tools

### ISSUE THREAD

**Issue by sockeqwe** [artem-zinnatullin/TheContext-Podcast#86]

**Comment by vanniktech** [comment]
Pretty interesting one!

At work we're using Teamcity and use Amazon S3 Agents and run a docker image on it. They boot up, will be used and then shutdown once they're done.

Tools that I use:

- [Refined Github](https://github.com/sindresorhus/refined-github) - it makes GitHub so much better
- [z](https://github.com/rupa/z) cd'ing into a directory was never easier `z rxjava` instead of `cd ~/tmp/RxJava2/`
- Local Gradle installation to not be dependent on project specific Gradle version and reuse Daemon between Terminal & AS
- [Oh my zsh](https://github.com/robbyrussell/oh-my-zsh) with [zsh-autosuggestions](https://github.com/zsh-users/zsh-autosuggestions)
- [ack](https://beyondgrep.com/) for searching everything
- [hub](https://hub.github.com/) for automatically creating PR from the command line for GitHub

I have modified all of my shortcuts from AS. I've not once found something that worked on someone else's machines. 😢 Luckily there's a super easy solution. `Quick Switch Theme` -> `Key Map` -> `Select yours`. You only have to import it and then can use it everywhere. 

I've also got aliases for almost everything. If I do something more than 5 times, there's an alias. They're synced via a Git repository in my home directory.

> Univeral Image Loader is deprecated but it's soo goood, it's really so good and it has everything you need ...

and I have a Tshirt of it, so I'm sold. :trollface: :trollface: 

Oh yes, please talk about agile. I also don't like it! 😼

**Comment by sockeqwe** [comment]
Thanks for your feedback and sharing your config!
Hub is new to me, looks useful, I will gove it a try.



Niklas Baudy <notifications@github.com> schrieb am Mi., 14. März 2018,
09:55:

**Comment by AndreSand** [comment]
Also at my work we use TeamCity and use Amazon linux-docker Agents :)

**Comment by vanniktech** [comment]
Yeah `z` does not care where you are and also allows fuzzy search.

Forgot to mention that my sublime 3 configuration is ["open source"](https://github.com/vanniktech/config-sublime3). I'm also working on making my home directory open but there are still a few things that need to be moved / reconsidered.

**Comment by artem-zinnatullin** [comment]
Oh nice, i also use Sublime Text 3 a lot because it's so much faster than IJ… The only problem i have with it is keybindings, would be so much comfortable to match ones from IJ

btw @vanniktech can you pls elaborate on Gradle daemon? IDE should reuse gradlew daemon if all properties are the same (very often it's not the case, i'm still having multiple daemons sometimes, need to figure it out)

**Comment by artem-zinnatullin** [comment]
Daaamn son, [Refined GitHub](https://github.com/sindresorhus/refined-github) is such a good find.  @vanniktech, many thanks! (works well in Firefox)

Btw, I totally forgot to mention [Paste 2](https://pasteapp.me/) for macOS. It's a copy-paste buffer history + search + pinning, it's foking amazing (one of the best apps i've bought). Highly recommend to at least get trial (and rebind shortcut to CMD + Shift + C)

**Comment by vanniktech** [comment]
I installed Gradle via homebrew `brew install gradle` and use it from the terminal as is. In AS I've configured to use it instead of the wrapper:

<img width="1149" alt="screen shot 2018-03-19 at 17 38 04" src="https://user-images.githubusercontent.com/5759366/37609106-7b96ec04-2b9c-11e8-84e3-4778d96caabb.png">

That way the Daemon will be reused between the terminal & AS. I often build the app in both and I'm also jumping between projects quite often with different Gradle versions.

Also I've disabled source generation. It's in the experimental tab. That way when you open a project AS won't start executing all of the gradle tasks to compile the default variant. (Takes forever on my machine and is much faster from Terminal).

Refined Github is pretty awesome yes. Even better it's open source and they're quite friendly with new features to make Github even better.

**Comment by artem-zinnatullin** [comment]
>I often build the app in both

Normally Daemons will be reused when you work with wrapper (often it's not the case unfortunately)

>I'm also jumping between projects quite often with different Gradle versions

Yeah, this case is problematic…

>Also I've disabled source generation. It's in the experimental tab. That way when you open a project AS won't start executing all of the gradle tasks to compile the default variant. (Takes forever on my machine and is much faster from Terminal).

Oh yeah that's a real issue, need more benchmarking though

**Comment by jshvarts** [comment]
Such a great episode as well as this discussion! Thanks all.
